src/abstractdataframe/abstractdataframe.jl

```diff
@@ -67,22 +67,48 @@ _names(df::AbstractDataFrame) = _names(index(df))
 
 Compat.hasproperty(df::AbstractDataFrame, s::Symbol) = haskey(index(df), s)
 
-function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol}; makeunique::Bool=false)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{Symbol};
+                 makeunique::Bool=false)
     rename!(index(df), vals, makeunique=makeunique)
     return df
 end
 
-function rename!(df::AbstractDataFrame, args...)
-    rename!(index(df), args...)
+function rename!(df::AbstractDataFrame, vals::AbstractVector{<:AbstractString};
+                 makeunique::Bool=false)
+    rename!(index(df), Symbol.(vals), makeunique=makeunique)
     return df
 end
+
+rename!(df::AbstractDataFrame, args::AbstractVector{Pair{Symbol,Symbol}}) =
+    rename!(index(df), args)
+
+function rename!(df::AbstractDataFrame,
+                 args::Union{AbstractVector{<:Pair{Symbol,<:AbstractString}},
```

Comment:
Is this needed because a `Union` triggers an ambiguity? I wouldn't have though so...

Reply:
What do you mean by "is this needed", as it can have two meanings - I will comment on both of them then:
1. why double `<:` is needed (in front of `Pair` and in front of `AbstrctString`) - this is needed because this is how parametric subtyping works
2. why I make a union of 7 different specifications? The reason is that writing `Pair{<:Union{Symbol,AbstractString}, <:Union{Symbol,AbstractString}}` would allow mixing `Symbols` and strings and we agreed that we want to disallow mixing. I could have written a macro that would iterate through all the options to generate appropriate methods but I decided that it would be too much magic and it is simpler just to make such a large union.

Reply:
Ah, right. That's really too bad we don't have a more concise way of doing this...